core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
 
   protected def spark = SparkSession.active
 
+  /**
+   * Issue 1144 - Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if both
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
+   * If not successful, we throw exception.
+   */
+  if(spark.conf.getOption(StaticSQLConf.SPARK_SESSION_EXTENSIONS.key)
```

Comment:
As a best practice, could you add an internal config in DeltaSQLConf to disable the check?